authz/providers/azure/rbac/rbac.go

```diff
@@ -443,33 +449,50 @@ func (a *AccessInfo) CheckAccess(request *authzv1.SubjectAccessReviewSpec) (*aut
 	}
 
 	// Fallback to managed namespace check
-	if !a.useManagedNamespaceResourceScopeFormat || a.clusterType != managedClusters {
-		klog.V(7).Infof("Checkaccess request is denied for user %s", checkAccessUsername)
-		return status, nil
-	}
+	managedNamespaceExists, managedNamespacePath := getManagedNameSpaceScope(request)
+	if a.useManagedNamespaceResourceScopeFormat && a.clusterType == managedClusters {
+		klog.V(7).Infof("Falling back to checking managed namespace scope for user %s", checkAccessUsername)
+		if !managedNamespaceExists {
+			klog.V(7).Infof(
+				"Skipping managed namespace check for user %s because subject access review is cluster scoped",
+				checkAccessUsername,
+			)
+			return status, nil
```

Comment:
this early return still doesn't look right to me.

I don't think that `a.useManagedNamespaceResourceScopeFormat && a.clusterType == managedClusters && !managedNamespaceExists` should imply an early return without checking Fleet RBAC.

Fleet RBAC is expected to work for cluster-scoped objects as well as namespace-scoped.

Reply:
I think this whole if block could be guarded by `a.useManagedNamespaceResourceScopeFormat && a.clusterType == managedClusters && managedNamespaceExists` ?

Reply:
I apparently missed this one.  Corrected it now.